File: java/src/test/java/com/haibo/yan/algorithm/array/TestZigzag.java
package com.haibo.yan.algorithm.array;

import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

@Test
public class TestZigzag {
    @DataProvider
    public Object[][] zigzagArray() {
        return new Object[][] {
                {"PAYPALISHIRING", 1, "PAYPALISHIRING"},
                {"PAYPALISHIRING", 2, "PYAIHRNAPLSIIG"},
                {"PAYPALISHIRING", 3, "PAHNAPLSIIGYIR"},
        };
    }

    @Test(dataProvider = "zigzagArray")
    public void testZigzagArray(String original, int row, String expected) {
        assertEquals(new Zigzag().convert(original, row), expected);
    }
}

File: javascript/test/jsarray/test-zigzag.js
'use strict'

var convert = require('../../js/jsarray/zigzag')
var assert = require('assert')

describe('Zigzag', function(){
    describe('#convert', function(){
        it('zigzag(PAYPALISHIRING, 3) should return PAHNAPLSIIGYIR', function(){
            assert.equal('PAHNAPLSIIGYIR', convert('PAYPALISHIRING', 3));
        })

        it('zigzag(ABC, 2) should return ACB', function(){
            assert.equal('ACB', convert('ABC', 2));
        })
    })
})